File: LC/[0487] Max Consecutive Ones II.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        mxl=0
        mxl_flipped=0
        mxl_unflipped=0

        for nn in nums:
            if nn==0:
                mxl_flipped = mxl_unflipped+1
                mxl_unflipped = 0
            else:
                mxl_flipped +=1
                mxl_unflipped +=1
            
            mxl=max(mxl, mxl_flipped, mxl_unflipped)
        
        return mxl
        

File: LC/[0487] Max Consecutive Ones II.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int count1=0, count2=0;
        //count1: upto now, the longest consecutive 1s ending here without flip
        //count2: upto now, the longest consecutive 1s ending here, with 1 flip
        int ret=0;
        for(auto x: nums){
            if(x==0){
                count2 = count1+1;
                count1=0;
            }else{
                ++count2;
                ++count1;
            }
            ret = max(ret, count2);
        }
        return ret;
    }
};